rt_utils/rtstruct_builder.py

```diff
@@ -29,14 +31,47 @@ def create_from(dicom_series_path: str, rt_struct_path: str) -> RTStruct:
         series_data = image_helper.load_sorted_image_series(dicom_series_path)
         ds = dcmread(rt_struct_path)
         RTStructBuilder.validate_rtstruct(ds)
+        RTStructBuilder.validate_rtstruct_series_references(ds, series_data)
 
-        # TODO create new frame of reference?
+        # TODO create new frame of reference? Right now we assume the last frame of reference created is suitable 
         return RTStruct(series_data, ds)
 
     @staticmethod
-    def validate_rtstruct(ds):
+    def validate_rtstruct(ds: Dataset):
+        """
+        Method to validate a dataset is a valid RTStruct containing the required fields
+        """
+
         if ds.SOPClassUID != SOPClassUID.RTSTRUCT or \
                 not hasattr(ds, 'ROIContourSequence') or \
                 not hasattr(ds, 'StructureSetROISequence') or \
                 not hasattr(ds, 'RTROIObservationsSequence'):
             raise Exception("Please check that the existing RTStruct is valid")
+
+    @staticmethod
+    def validate_rtstruct_series_references(ds: Dataset, series_data: List[Dataset]):
+        """
+        Method to validate RTStruct only references dicom images found within the input series_data
+        """
+        # Blame DICOM for this nesting
+        for refd_frame_of_ref in ds.ReferencedFrameOfReferenceSequence:
```

Comment:
lmk if i can make this cleaner. Also lmk if you have better ideas for method names